### PR TITLE
Respect deprecated.legacy-catalog-roles=true in SHOW ROLES

### PR DESCRIPTION
**Pull request by kokosing**
Respect deprecated.legacy-catalog-roles=true in SHOW ROLES

### CODE REVIEW

**Comment by kokosing** [comment]
Fixes https://github.com/trinodb/trino/issues/9485

File: core/trino-main/src/main/java/io/trino/execution/CreateMaterializedViewTask.java [hunk 1]
```diff
@@ -53,13 +54,15 @@
...
+    private final boolean legacyCatalogRoles;
 
     @Inject
-    public CreateMaterializedViewTask(SqlParser sqlParser, GroupProvider groupProvider, StatsCalculator statsCalculator)
+    public CreateMaterializedViewTask(SqlParser sqlParser, GroupProvider groupProvider, StatsCalculator statsCalculator, FeaturesConfig featuresConfig)
     {
         this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
         this.groupProvider = requireNonNull(groupProvider, "groupProvider is null");
         this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
+        legacyCatalogRoles = requireNonNull(featuresConfig, "featuresConfig is null").isLegacyCatalogRoles();
```
Comment (ksobolew) [inline core/trino-main/src/main/java/io/trino/execution/CreateMaterializedViewTask.java:65]: nit:

```suggestion
        this.legacyCatalogRoles = requireNonNull(featuresConfig, "featuresConfig is null").isLegacyCatalogRoles();
```

(Just for consistency)

**Review by ksobolew — APPROVED** [review]
Such a silly bug... I wonder how many similar ones there are.

**Comment by guyco33** [comment]
Just verified that it fixes #9485
Thanks @kokosing

**Comment by martint** [comment]
I haven’t looked at the code in depth, but I noticed we’re sprinkling knowledge about legacyCatalogRoles everywhere. We shouldn’t need to do that - there’s probably a better way to get that information where it’s needed.

**Comment by kokosing** [comment]
> I haven’t looked at the code in depth, but I noticed we’re sprinkling knowledge about legacyCatalogRoles everywhere.

The issue is already in the code. Whenever you want to use `Analyzer` you need to have all its dependencies at hand. So classes need to have `GroupProvider` only for that. I think it would be nice to somehow handle dependency injection at query level. I spent a bit of time to introduce `AnalyzerFactory` that would have not require to provide things that do not changed per query in order to create `Analyzer`, but I was not satisfied with the end shape. Also it was too big change to introduce along with a bug fix.

> there’s probably a better way to get that information where it’s needed.

In https://github.com/trinodb/trino/pull/9496 I used hidden session property. However, it changes the product so it is a bit too much for a plain bugfix. On the other hand that session property could be useful too. Please take a look.

**Review by gaurav8297 — APPROVED** [review]

**Comment by kokosing** [comment]
Closing in favor of #9496

**Pull request closed without merging**